### ISSUE TITLE
ERR_REQUIRE_ESM error after update nanoid 5.0.7

### ISSUE THREAD

**Issue by Fatihkrty** [mgcrea/fastify-session#32]
Hi,

I get this error after the update. I remember getting this error before in a different project for the nanoid 4 version. 
It's a problem for projects using CommonJs.

[Issue](https://github.com/ai/nanoid/issues/365)

[Stackoverflow Solution](https://stackoverflow.com/a/73191957/18453029)

[Nanoid Changelog v4.0](https://github.com/ai/nanoid/blob/main/CHANGELOG.md#40)

```
/home/fatih/Desktop/points/backend/node_modules/ts-node/dist/index.js:851
            return old(m, filename);
                   ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/fatih/Desktop/points/backend/node_modules/nanoid/index.js from /home/fatih/Desktop/points/backend/node_modules/@mgcrea/fastify-session/dist/index.cjs not supported.
Instead change the require of index.js in /home/fatih/Desktop/points/backend/node_modules/@mgcrea/fastify-session/dist/index.cjs to a dynamic import() which is available in all CommonJS modules.
    at require.extensions.<computed> [as .js] (/home/fatih/Desktop/points/backend/node_modules/ts-node/dist/index.js:851:20)
    at Object.<anonymous> (/home/fatih/Desktop/points/backend/node_modules/@mgcrea/fastify-session/dist/index.cjs:120:21)
    at require.extensions.<computed> [as .js] (/home/fatih/Desktop/points/backend/node_modules/ts-node/dist/index.js:851:20) {
  code: 'ERR_REQUIRE_ESM'
}

```

**Comment by mgcrea** [comment]
Did revert to 4.x, should work with the just released v2.4.1

**Comment by reesemartin** [comment]
Important to note that nanoid's guidance is to use 3.x for commonjs support which they will continue to support [as indicated in their install instructions](https://github.com/ai/nanoid?tab=readme-ov-file#install). For anyone who is using 3.x based on that guidance, his package using 4.x will require a resolution override.
